lib/app_profiler/profile.rb

```diff
@@ -79,11 +96,14 @@ def path
         mode,
         id,
         Socket.gethostname,
-      ].compact.join("-") << ".json"
+      ].compact.join("-") << format
 
       raise UnsafeFilename if /[^0-9A-Za-z.\-\_]/.match?(filename)
 
       AppProfiler.profile_root.join(filename)
     end
   end
+
+  Profile = AbstractProfile
+  deprecate_constant :Profile
```

Comment:
@gmcgibbon I couldn't get the [rails one](https://api.rubyonrails.org/classes/ActiveSupport/Deprecation/DeprecatedConstantAccessor.html#method-i-deprecate_constant) to work as expected, when I tried to test it out it would give the deprecation warning, but then also cause an error. I'm just using the basic ruby one instead as I couldn't figure out why the other one kept exploding.

Reply:
Does that happen in a rails app? Check to make sure the deprecation behaviour in the application is set to log, if you want it to log. I think the fact that it was raising is dependent on the app's config.

Reply:
Ok to be clear, here is the error I'm getting if I try and use the old constant:

```
DEPRECATION WARNING: AppProfiler::Profile is deprecated! Use AbstractProfile instead. (called from <top (required)> at /opt/rubies/3.2.1/lib/ruby/gems/3.2.0/gems/irb-1.6.2/exe/irb:11)
/Users/dalehamel/.gem/ruby/3.2.1/gems/activesupport-5.2.4.3/lib/active_support/inflector/methods.rb:283:in `const_get': uninitialized constant AbstractProfile (NameError)

          constant.const_get(name)
                  ^^^^^^^^^^
```

So it gives a warning, then it shits the bed trying to find the constant.

Here's how I am using it:

```
...

107   include ActiveSupport::Deprecation::DeprecatedConstantAccessor
108   deprecate_constant 'Profile', 'AbstractProfile', deprecator: ActiveSupport::Deprecation.new

```

It is fine to just use `AbstractProfile` directly.